src/encoder.rs

```diff
@@ -40,16 +40,28 @@ pub fn encode<T: Instructions>(ast: &Block, loop_unroll: u64) -> String {
         .join("\n")
 }
 
-pub fn encode_revert_unreachable<T: Instructions>(ast: &Block, loop_unroll: u64) -> String {
+pub fn encode_revert_unreachable<T: Instructions>(
+    ast: &Block,
+    loop_unroll: u64,
+    counterexamples: &[Expression],
+) -> (String, Vec<String>) {
     let mut encoder = Encoder::<T>::default();
     encoder.encode(ast, loop_unroll);
     encoder.encode_revert_unreachable();
-    encoder
-        .output
+
+    let encoded_counterexamples = counterexamples
         .iter()
-        .map(|s| s.as_smt())
-        .collect::<Vec<_>>()
-        .join("\n")
+        .map(|expr| encoder.encode_expression(expr).pop().unwrap().name)
```

Comment:
what I meant earlier was, what if you ask for a cex for `f(2)` where `f` returns two numbers? this would only accept the last

Reply:
Ah ok! I think it's fine to only support single expressions for now, do you agree? I can add an assertion about that.

Reply:
yep, just wanted to make sure it was intended vs bug